menu/drivers/xmb.c

```diff
@@ -3009,39 +3009,57 @@ static void xmb_frame(void *data, video_frame_info_t *video_info)
    menu_display_rotate_z(&rotate_draw, video_info);
    menu_display_blend_begin(video_info);
 
-   if (xmb->savestate_thumbnail)
-      xmb_draw_thumbnail(video_info,
-            xmb, &coord_white[0], width, height,
-            xmb->margins_screen_left * scale_mod[5]
-            + xmb->icon_spacing_horizontal +
-            xmb->icon_spacing_horizontal * 4 - xmb->icon_size / 4,
-            xmb->margins_screen_top + xmb->icon_size + xmb->savestate_thumbnail_height * scale_mod[4],
-            xmb->savestate_thumbnail_width, xmb->savestate_thumbnail_height,
-            xmb->savestate_thumbnail);
-   else if (xmb->thumbnail
-      && !string_is_equal(xmb_thumbnails_ident(),
-         msg_hash_to_str(MENU_ENUM_LABEL_VALUE_OFF)))
+   /* Do not draw the thumbnail if there is no space */
+   if ((50 + xmb->margins_screen_top + xmb->icon_size) < (float)(height) &&
+      ((50 + xmb->margins_screen_left * scale_mod[5] + xmb->icon_spacing_horizontal +
+               xmb->icon_spacing_horizontal*4 - xmb->icon_size / 4)< (float)(width)))
    {
+      if (xmb->savestate_thumbnail)
+         xmb_draw_thumbnail(video_info,
+               xmb, &coord_white[0], width, height,
+               xmb->margins_screen_left * scale_mod[5]
+               + xmb->icon_spacing_horizontal +
+               xmb->icon_spacing_horizontal * 4 - xmb->icon_size / 4,
+               xmb->margins_screen_top + xmb->icon_size + xmb->savestate_thumbnail_height * scale_mod[4],
+               xmb->savestate_thumbnail_width, xmb->savestate_thumbnail_height,
+               xmb->savestate_thumbnail);
+      else if (xmb->thumbnail
+         && !string_is_equal(xmb_thumbnails_ident(),
+            msg_hash_to_str(MENU_ENUM_LABEL_VALUE_OFF)))
+      {
 #ifdef XMB_DEBUG
       RARCH_LOG("[XMB thumbnail] width: %.2f, height: %.2f\n", xmb->thumbnail_width, xmb->thumbnail_height);
       RARCH_LOG("[XMB thumbnail] w: %.2f, h: %.2f\n", width, height);
 #endif
 
-      /* Limit thumbnail height to screen height + margin. */
-      if( xmb->margins_screen_top + xmb->icon_size + xmb->thumbnail_height * scale_mod[4] >= 
-         (float)(height * 0.96) )
-      {
-         thumb_width = xmb->thumbnail_width *
-            (((float)(height * 0.96) - xmb->margins_screen_top - xmb->icon_size) /
-               (xmb->thumbnail_height * scale_mod[4]));
-         thumb_height = xmb->thumbnail_height *
-            (((float)(height * 0.96) - xmb->margins_screen_top - xmb->icon_size) /
-               (xmb->thumbnail_height * scale_mod[4]));
-      }
-      else
-      {
+         /* Limit thumbnail height to screen height + margin. */
+
          thumb_width = xmb->thumbnail_width;
          thumb_height = xmb->thumbnail_height;
+
+         if ( xmb->margins_screen_top + xmb->icon_size + xmb->thumbnail_height * scale_mod[4] >=
+            (float)(height * 0.96) )
```

Comment:
`0.96` looks like a [Magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) ... Perhaps stick that in a border_scale variable float or similar?

Also, if you make it `0.96f`, then you may not need the casing.

Reply:
i'll try to clean it up